scripts/language_model/cache_language_model.py

```diff
@@ -0,0 +1,202 @@
+"""
+Neural Cache Language Model
+===================
+This example shows how to build a neural cache language model based on
+pre-trained word-level language model on WikiText-2 with Gluon NLP Toolkit.
+
+We implement the neural cache language model proposed in the following work.
+@article{grave2016improving,
+  title={Improving neural language models with a continuous cache},
+  author={Grave, Edouard and Joulin, Armand and Usunier, Nicolas},
+  journal={ICLR},
+  year={2017}
+}
+"""
+
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import argparse
+import time
+import math
+import os
+import sys
+import mxnet as mx
+import gluonnlp as nlp
+
+curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
+sys.path.append(os.path.join(curr_path, '..', '..'))
+
+
+parser = argparse.ArgumentParser(description=
+                                 'MXNet Neural Cache Language Model on Wikitext-2.')
+parser.add_argument('--bptt', type=int, default=2000,
+                    help='sequence length')
+parser.add_argument('--save', type=str, default='awd_lstm_lm_1150',
+                    help='name of the pretrained language model')
```

Comment:
this argument is later used to load user's model, which is counterintuitive to me. currently the related options look like
```
--save: name of the pretrained language model
--user_model: whether to apply cache model to user pre-trained model
```
instead, we can use the following options:
```
--model-name: name of the pretrained language model (and explain where to find the options)
--param-file: file name/path of the weights (and explain that if this is not specified, script will use/download pretrained model)
```